core/src/main/java/net/opentsdb/query/AbstractQueryPipelineContext.java

```diff
@@ -59,7 +52,7 @@
  * 
  * <b>Invariants:</b>
  * <ul>
- * <li>Each {@link ExecutionGraphNode} must have a unique ID within the graph.</li>
+ * <li>Each {@link ExecutionGraphNode} must have a unique ID within the plan.graph().</li>
```

Comment:
Are these invariants checked statically (can't find the code yet), or do they cause run-time errors?

Reply:
Yes, throws a runtime error in the new `DefaultQueryPlanner` when it tries to build the graph.